src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -5,207 +5,197 @@
+                if (_requestMessage.Content != null)
+                {
+                    return HttpHeadersToStringDictionary(_requestMessage.Headers).Concat(HttpHeadersToStringDictionary(_requestMessage.Content.Headers));
+                }
+
+                return HttpHeadersToStringDictionary(_requestMessage.Headers).Concat(_contentHeaderValueCache);
             }
         }
 
-        #region Properties
-
-        /// <summary> Returns the request url. </summary>
+        /// <summary>
+        /// Gets or sets the request url.
+        /// </summary>
         public override Uri Url
         {
             get
             {
-                return this.requestUrl;
+                return _requestUrl;
             }
-
             set
             {
                 throw new NotSupportedException();
             }
         }
 
-        /// <summary> Returns the method for this request. </summary>
+        /// <summary>
+        /// Gets or sets the method for this request.
+        /// </summary>
         public override string Method
         {
             get
             {
-                return this.effectiveHttpMethod;
+                return _effectiveHttpMethod;
             }
-
             set
             {
                 throw new NotSupportedException();
             }
         }
 
-        /// <summary> Returns the collection of request headers. </summary>
-        public override IEnumerable<KeyValuePair<string, string>> Headers
+        /// <summary>
+        ///  Gets or set the credentials for this request.
+        /// </summary>
+        public override ICredentials Credentials
         {
             get
             {
-                List<KeyValuePair<string, string>> headerValues = new List<KeyValuePair<string, string>>(this.httpRequest.Headers.Count);
-                foreach (string headerName in this.httpRequest.Headers.AllKeys)
-                {
-                    string headerValue = this.httpRequest.Headers[headerName];
-                    headerValues.Add(new KeyValuePair<string, string>(headerName, headerValue));
-                }
-
-                return headerValues;
+                return _handler.Credentials;
             }
-        }
-
-        /// <summary> Returns the underlying HttpWebRequest </summary>
-        public System.Net.HttpWebRequest HttpWebRequest
-        {
-            get
+            set
             {
-                return this.httpRequest;
+                _handler.Credentials = value;
             }
         }
 
-        /// <summary>
-        /// Gets or set the credentials for this request.
-        /// </summary>
-        public override System.Net.ICredentials Credentials
-        {
-            get { return this.httpRequest.Credentials; }
-            set { this.httpRequest.Credentials = value; }
-        }
+
+#if !(NETCOREAPP1_0 || NETCOREAPP2_0)
 
         /// <summary>
         /// Gets or sets the timeout (in seconds) for this request.

```

Comment:
Consider updating the description of this prop so it's clear how it's different from `ReadWriteTimeout` prop